common/lib/common-utils/src/assert.ts

```diff
@@ -0,0 +1,16 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * A browser friendly version of the node assert library. Use this instead of the 'assert' package, which has a big
+ * impact on bundle sizes.
+ * @param condition - The condition that should be truthy, if the condition is falsy an error will be thrown.
+ * @param message - The message to include in the error when the condition does not hold
+ */
+ export function assert(condition: any, message: string) {
```

Comment:
Can condition be boolean (i.e. force conversion at call site)?
Also can it be strict assert? I.e. 

export function assert(condition: boolean, message: string): asserts condition is true {
}

Reply:
message should be optional

Reply:
Yup, good suggestion. Originally had that, but wanted to minimize the next change so fell back to truthy/falsy, but I think it is the right change while I am going through all the asserts